src/runtime/test_case_runner.js

```diff
@@ -199,7 +212,13 @@ export default class TestCaseRunner {
 
   async runSteps() {
     await Promise.each(this.testCase.pickle.steps, async step => {
+      // TODO handle failure
+      await this.runStepHooks(this.beforeStepHookDefinitions, {
+        sourceLocation: this.testCaseSourceLocation,
+        pickle: this.testCase.pickle,
+      })
```

Comment:
I think this should be within the call to `this.aroundTestStep`. That function emits the events `test-step-started/finished`. I think these before/after step hooks should actually be thought of as part of the step. We would run the beforeStep/step/afterStep and merge the results of each. The step fails if the beforeStep, step, or afterStep fails. We would also not run the step if the beforeStep fails.